tmail-backend/event-bus-redis/src/main/java/org/apache/james/events/CleanRedisEventBusService.java

```diff
@@ -0,0 +1,147 @@
+            totalBindings.incrementAndGet();
+        }
+
+        void increaseDanglingBindingsCount(long count) {
+            danglingBindings.addAndGet(count);
+        }
+
+        void increaseCleanedBindingsCount(long count) {
+            cleanedBindings.addAndGet(count);
+        }
+
+        public long getTotalBindings() {
+            return totalBindings.get();
+        }
+
+        public long getDanglingBindings() {
+            return danglingBindings.get();
+        }
+
+        public long getCleanedBindings() {
+            return cleanedBindings.get();
+        }
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CleanRedisEventBusService.class);
+
+    private final RedisSetReactiveCommands<String, String> redisSetCommands;
+    private final RedisPubSubReactiveCommands<String, String> redisPubSubCommands;
+    private final RedisKeyReactiveCommands<String, String> redisKeyCommands;
+    private final RoutingKeyConverter routingKeyConverter;
+    private Context context;
+
+    @Inject
+    @Singleton
+    public CleanRedisEventBusService(RedisEventBusClientFactory redisEventBusClientFactory,
+                                     RoutingKeyConverter routingKeyConverter) {
+        this.redisSetCommands = redisEventBusClientFactory.createRedisSetCommand();
+        this.redisPubSubCommands = redisEventBusClientFactory.createRedisPubSubCommand();
+        this.redisKeyCommands = redisEventBusClientFactory.createRedisKeyCommand();
+        this.routingKeyConverter = routingKeyConverter;
+    }
+
+    public Mono<Void> cleanUp() {
+        this.context = new Context();
+
+        return listActiveRedisChannels()
+            .collectList()
+            .flatMap(activeChannels -> listRedisSetKeys()
+                .filter(this::isEventBusBindingKey)
+                .flatMap(key -> listElementsOfSet(key)
+                        .doOnNext(channel -> context.increaseScannedBindingsCount())
+                        .filter(channel -> !activeChannels.contains(channel))
+                        .collectList()
+                        .doOnNext(inactiveChannels -> context.increaseDanglingBindingsCount(inactiveChannels.size()))
+                        .flatMap(deleteBindingsToInactiveChannels(key)),
+                    ReactorUtils.DEFAULT_CONCURRENCY)
+                .then())
+            .doOnSuccess(any -> mdcLogWithContext()
+                .log(logger -> logger.info("Clean dangling eventbus bindings in Redis successfully")))
+            .doOnError(e -> mdcLogWithContext()
+                .log(logger -> logger.error("Failed cleaning dangling eventbus bindings in Redis", e)));
+    }
+
+    private Flux<String> listActiveRedisChannels() {
+        return redisPubSubCommands.pubsubChannels();
+    }
+
+    private Flux<String> listRedisSetKeys() {
+        return redisKeyCommands.scan(ScanCursor.INITIAL, KeyScanArgs.Builder.type("set"))
+            .flatMapIterable(KeyScanCursor::getKeys);
+    }
```

Comment:
Testing on staging, I spotted there is a bug where Redis only returns 10/11 elements as a first batch. I would need to do some iteration to get all the elements here (with a test)